src/array.ts

```diff
@@ -82,8 +83,13 @@ export class ArrayType<T extends IAnyType> extends BaseType<Array<T["InputType"]
   instantiate(snapshot: this["InputType"] | undefined, context: TreeContext, parent: IStateTreeNode | null): this["InstanceType"] {
     const array = new QuickArray<T>(this, parent, context);
     if (snapshot) {
-      array.push(...snapshot.map((element) => this.childrenType.instantiate(element, context, array)));
+      let instances = snapshot.map((element) => this.childrenType.instantiate(element, context, array));
+      if (this.childrenType instanceof SafeReferenceType && this.childrenType.options?.acceptsUndefined === false) {
+        instances = instances.filter((instance) => instance !== null && instance !== undefined);
```

Comment:
nit: similar to below, you could also just write `instance != undefined`, but in this case using strict inequality there's no redundancy, unlike below :)